### PR TITLE
Fix input focus for preexisting clients

### PR DESCRIPTION
**Pull request by t-wissmann**
This correctly puts the input focus to clients that exist before hlwm.
This is a bug that was brought to the surface by PR #1403, because now,
hlwm only calls XSetInputFocus() if the focus actually changes.

The solution now is: the decoration window is 'unmapped' if and only if
the client window itself is 'unmapped'. This is true already for all
clients that show up during the hlwm execution, but wasn't true for
clients existing before. With this property, making the client visible
(in Client::set_visible()) will actually set the client to the 'mapped'
state, triggering a MapNotify event, whose handler then calls
XSetInputFocus() if necessary.

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -204,6 +204,9 @@ void Client::make_full_client() {
         // point, we did not call XSelectInput() yet, so the above
         // XReparentWindow does not trigger an event for the window.
         visible_ = false;
+        // we also unmap the window such that the decoration window is
+        // in 'mapped' state f and only if the window itself is so.
```
Comment (blarz) [inline src/client.cpp:208]: "i" missing

**Review by blarz — COMMENTED** [review]
Just a typo

File: src/client.cpp (hunk 1 above)
Comment (t-wissmann) [reply src/client.cpp]: [x] fixed

Comment (blarz) [reply src/client.cpp]: Thanks!

**Review by blarz — APPROVED** [review]

**Pull request merged**